### PR TITLE
Add Gyro

### PR DESCRIPTION
**Pull request by noahheller**

### CODE REVIEW

**Comment by noahheller** [comment]
PR depends on Logging + AdvantageKit PR

File: src/main/java/org/usfirst/frc4048/common/gyro/RealGyroIO.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package org.usfirst.frc4048.common.gyro;
+
+import com.kauailabs.navx.frc.AHRS;
+
+public class RealGyroIO implements GyroIO {
+    private final ThreadedGyro gyro;
+    private double angleOffset = 0;
+
+    public RealGyroIO() {
+        this.gyro = new ThreadedGyro(new AHRS());
```
Comment (armadilloz) [inline src/main/java/org/usfirst/frc4048/common/gyro/RealGyroIO.java]: this does not look right - why would the logging system be initializing the gyro thread?

File: src/main/java/org/usfirst/frc4048/common/gyro/ThreadedGyro.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    private final AtomicLong gyroOffset = new AtomicLong();
+    private final ScheduledExecutorService executor;
+
+    public ThreadedGyro(AHRS gyro) {
+        this.gyro = gyro;
+        this.lastGyro = new AtomicLong((Double.doubleToLongBits(0)));
+        this.executor = Executors.newScheduledThreadPool(1);
+    }
+
+    public void start(){
```
Comment (armadilloz) [inline src/main/java/org/usfirst/frc4048/common/gyro/ThreadedGyro.java:27]: We can probably add `stop` method (though I am not sure where we would call it from)

File: src/main/java/org/usfirst/frc4048/common/gyro/RealGyroIO.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        gyro.start();
+    }
+
+    @Override
+    public void setAngleOffset(double offset) {
+        this.angleOffset = offset;
```
Comment (armadilloz) [inline src/main/java/org/usfirst/frc4048/common/gyro/RealGyroIO.java]: Why do we need this here? You can cache this in the `ThreadedGyro` if you want to save hardware access

**Review by armadilloz — CHANGES_REQUESTED** [review]

File: src/main/java/org/usfirst/frc4048/common/gyro/RealGyroIO.java (hunk 3 above)
Comment (noahheller) [reply src/main/java/org/usfirst/frc4048/common/gyro/RealGyroIO.java]: done

File: src/main/java/org/usfirst/frc4048/common/gyro/RealGyroIO.java (hunk 1 above)
Comment (noahheller) [reply src/main/java/org/usfirst/frc4048/common/gyro/RealGyroIO.java]: The threaded gyro is Hardware Specific so I initialize it in the IO interface as if it was a motor for the intake or something like that.

File: src/main/java/org/usfirst/frc4048/common/gyro/ThreadedGyro.java (hunk 2 above)
Comment (noahheller) [reply src/main/java/org/usfirst/frc4048/common/gyro/ThreadedGyro.java:27]: done (me either, but it is good to have)

File: src/main/java/org/usfirst/frc4048/common/gyro/RealGyroIO.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
+package org.usfirst.frc4048.common.gyro;
+
+import com.kauailabs.navx.frc.AHRS;
+
+public class RealGyroIO implements GyroIO {
+    private final ThreadedGyro gyro;
+
+    public RealGyroIO(ThreadedGyro gyro) {
+        this.gyro = gyro;
+        gyro.start();
```
Comment (armadilloz) [inline src/main/java/org/usfirst/frc4048/common/gyro/RealGyroIO.java]: move to configuration part (robot container)

**Review by armadilloz — APPROVED** [review]

**Pull request merged**